ts/features/wallet/onboarding/cobadge/screens/search/__test__/SearchAvailableCoBadgeScreen.test.tsx

```diff
@@ -0,0 +1,308 @@
+    store.dispatch(searchUserCoBadge.success(malformedData));
+    expect(isTimeoutScreen(testComponent)).toBe(true);
+  });
+  it("With no card found, single abi, should render CoBadgeKoSingleBankNotFound", () => {
+    const { store, testComponent } = getSearchAvailableCoBadgeScreen();
+    store.dispatch(searchUserCoBadge.request(abiTestId));
+    store.dispatch(searchUserCoBadge.success(noCardResponse));
+    expect(isCoBadgeKoSingleBankNotFoundScreen(testComponent)).toBe(true);
+
+    const searchAllBankButton = testComponent.queryByText(
+      I18n.t("global.buttons.continue")
+    );
+    expect(searchAllBankButton).toBeTruthy();
+
+    if (searchAllBankButton !== null) {
+      fireEvent.press(searchAllBankButton);
+      // After press the continue button, start the search for all abi
+      expect(isLoadingScreen(testComponent)).toBe(true);
+      // Simulate the response with no card
+      store.dispatch(searchUserCoBadge.success(noCardResponse));
+      // The user will see the Ko screen not found for all abi search
+      expect(isCoBadgeKoNotFoundScreen(testComponent)).toBe(true);
+    }
+  });
+  it("With no card found, all abi, should render CoBadgeKoNotFound", () => {
+    const { store, testComponent } = getSearchAvailableCoBadgeScreen();
+    store.dispatch(searchUserCoBadge.request(undefined));
+    store.dispatch(searchUserCoBadge.success(noCardResponse));
+    expect(isCoBadgeKoNotFoundScreen(testComponent)).toBe(true);
+  });
+  describe("Search request metadata pending behaviour", () => {
+    [undefined, abiTestId].map(abi =>
+      it(`With at least one pending in search metadata request, abi=${abi}, should render CoBadgeKoServiceError`, () => {
+        const { store, testComponent } = getSearchAvailableCoBadgeScreen();
+        store.dispatch(searchUserCoBadge.request(undefined));
+        store.dispatch(
+          searchUserCoBadge.success(
+            withSearchRequestMetadata([searchRequestMetaPending])(
+              noCardResponse
+            )
+          )
+        );
+        expect(isTimeoutScreen(testComponent)).toBe(true);
+        store.dispatch(
+          searchUserCoBadge.success(
+            withSearchRequestMetadata([
+              searchRequestMetaPending,
+              searchRequestMetaPending
+            ])(noCardResponse)
+          )
+        );
+        expect(isTimeoutScreen(testComponent)).toBe(true);
+        store.dispatch(
+          searchUserCoBadge.success(
+            withSearchRequestMetadata([
+              searchRequestMetaPending,
+              searchRequestMetaOK
+            ])(noCardResponse)
+          )
+        );
+        expect(isTimeoutScreen(testComponent)).toBe(true);
+        store.dispatch(
+          searchUserCoBadge.success(
+            withSearchRequestMetadata([
+              searchRequestMetaPending,
+              searchRequestMetaKO
+            ])(noCardResponse)
+          )
+        );
+        expect(isTimeoutScreen(testComponent)).toBe(false);
```

Comment:
This case is already tested in line 225.
Do you think that make sense move the assertion below? 
```suggestion
```

Reply:
Yes, you are right, removed in 0448766